include/ginkgo/core/base/array.hpp

```diff
@@ -121,14 +121,27 @@ class const_array_view {
      *
      * @return the number of elements in the array view
      */
-    size_type get_num_elems() const noexcept { return num_elems_; }
+    size_type size() const noexcept { return num_elems_; }
+
+    /** @copydoc size() */
+    [[deprecated("use size() instead")]] size_type get_size() const noexcept
```

Comment:
We need to deprecate the old function:
```suggestion
    [[deprecated("use size() instead")]] size_type get_num_elems() const noexcept
```
Same for the device_matrix_data

Reply:
oops, right, thanks!